vllm/v1/engine/output_processor.py

```diff
@@ -79,6 +88,89 @@ def from_new_request(
             log_stats=log_stats,
         )
 
+    def make_request_output(
+        self,
+        new_token_ids: List[int],
+        finish_reason: Optional[FinishReason],
+        stop_reason: Union[int, str, None],
+    ) -> Optional[RequestOutput]:
+
+        finished = finish_reason is not None
+        output_kind = self.output_kind
+        final_only = output_kind == RequestOutputKind.FINAL_ONLY
+
+        # In follow up, we will switch to invariant where EngineCore
+        # does not stream partial prefills.
+        if not finished and (self.is_prefilling or final_only):
+            # Only the final output is required in FINAL_ONLY mode.
+            return None
+
+        def new_request_output(request_id: str) -> RequestOutput:
+            return self._new_request_output(request_id, finished)
+
+        def append_completion_output(
+                request_output: RequestOutput) -> RequestOutput:
+            request_output.outputs.append(
+                self._new_completion_output(new_token_ids, finish_reason,
+                                            stop_reason))
+            return request_output
```

Comment:
I'm not sure we need this method, could we just create the completion output here and pass it directly to `parent_req.make_request_output`?

Reply:
Ok, done :+1: